src/components/common/FileSelect/FileSelect.js

```diff
@@ -25,76 +38,47 @@ class FileSelect extends Component {
     label: PropTypes.string,
     accept: PropTypes.array,
     multiple: PropTypes.bool,
-    fileSize: PropTypes.number,
-    input: PropTypes.object
+    maxFileSize: PropTypes.number,
+    input: PropTypes.object,
+    aspectRatio: PropTypes.number,
+    maxFiles: PropTypes.number
   }
 
   constructor (props) {
     super(props)
     this.state = {
-      isLoading: false,
-      isLoaded: false,
-      error: null,
-      accept: props.accept || ACCEPT_IMAGES_AND_DOC,
-      files: []
-    }
-  }
-
-  componentWillMount () {
-    this.setState({isLoaded: !!this.props.value})
-  }
-
-  getImageDimensions (file) {
-    return new Promise(resolve => {
-      const _URL = window.URL || window.webkitURL
-      const img = new Image()
-      img.onload = () => {
-        resolve({
-          width: img.width,
-          height: img.height
-        })
+      files: new Immutable.Map(),
+      fileCollection: new FileCollection(),
+      config: {
+        accept: props.accept || ACCEPT_ALL,
+        maxFileSize: props.maxFileSize || DEFAULT_MAX_FILE_SIZE,
+        aspectRatio: props.aspectRatio || DEFAULT_ASPECT_RATIO,
+        maxFiles: props.maxFiles || DEFAULT_MAX_FILES
       }
-      img.src = _URL.createObjectURL(file)
-    })
-  }
-
-  async checkImageFile (file) {
-    // parse size
-    if (file.size > DEFAULT_MAX_FILE_SIZE) {
-      throw new Error('Exceeded the maximum file size (Limit: 2Mb)')
-    }
-
-    // parse dimensions
-    const {width, height} = await this.getImageDimensions(file)
-    if (width === 0 || height === 0) {
-      throw new Error('Wrong image sizes')
     }
-
-    // parse ratio
-    const ratio = width / height
-    if (ratio > DEFAULT_ASPECT_RATIO || ratio < 1 / DEFAULT_ASPECT_RATIO) {
-      throw new Error('Wrong image aspect ratio (Limit from 1:2 to 2:1)')
-    }
-
-    // all tests passed
-    return file
   }
 
-  parseFile (file = {}) {
-    if (!file.name) {
-      throw new Error('No file name')
-    }
-
-    if (!this.state.accept.includes(file.type)) {
-      throw new Error('Wrong file type')
+  componentDidMount () {
+    const input = this.props.input
+    if (input && input.value) {
+      this.loadCollection(this.props.input.value)
     }
+  }
 
-    if (ACCEPT_IMAGES.includes(file.type)) {
-      return this.checkImageFile(file)
-    }
-    if (ACCEPT_DOCS.includes(file.type)) {
-      return file
+  async loadCollection (hash) {
+    const data = await ipfs.get(hash)
+    let fileCollection = new FileCollection({
+      hash: hash,
+      uploaded: data && data.links.length
+    })
+    if (data && data.links) {
+      for (const item of data.links) {
+        fileCollection = fileCollection.add(FileModel.createFromLink(item))
+      }
     }
+    this.setState({
+      fileCollection
+    })
   }
 
   async addToIPFS (name, rawData) {
```

Comment:
remove unused medthods

Reply:
also addToIPFSAsFile and addToIPFSAsObject

Reply:
Done